test/gurka/test_bidir_search.cc

```diff
@@ -40,22 +40,15 @@ TEST(StandAlone, exhaust_reverse_search) {
     }
   });
 
-  // Without extending search, the route should fail
+  // Without extending search, the route should not fail due to settting allow_restricted_thru
+  // to true on the second pass
   map.config.put("thor.extended_search", false);
-  EXPECT_THROW(
-      {
-        try {
-          auto result = gurka::do_action(valhalla::Options::route, map, {"A", "F"}, "auto", {});
-        } catch (const std::exception& e) {
-          EXPECT_STREQ("No path could be found for input", e.what());
-          throw;
-        }
-      },
-      std::exception);
+  auto result = gurka::do_action(valhalla::Options::route, map, {"A", "F"}, "auto", {});
```

Comment:
hmmm, why is not failing to find a route anymore with your change? This is a simple `not_thru` case and the previous open edge does not have any restriction on it. I thought this case would still throw a no-route exception since there's no other restriction involved.

Reply:
@mandeepsandhu in the reverse search not_thru is not considered anymore on the 2nd pass